p2p/3.foundry/README.md

```diff
@@ -1,111 +1,238 @@
-    </a>
-    <a href="https://discord.com/invite/Yqq2ADGDS7">
-        <img src="https://img.shields.io/badge/Discord-7289DA?style=for-the-badge&logo=discord&logoColor=white">
-    </a>
-</p>
-<p align=center>
-    <a href="https://www.poc-innovation.fr/">
-        <img src="https://img.shields.io/badge/WebSite-1a2b6d?style=for-the-badge&logo=GitHub Sponsors&logoColor=white">
-    </a>
-</p>
-
-> :rocket: Don't hesitate to follow us on our different networks, and put a star 🌟 on `PoC's` repositories.
+# Workshop 3 - Foundry, a toolkit for Ethereum application development
+
+✔ How to use foundry to test your contracts.
+
+✔ How to use Foundry to deploy your contracts on a local blockchain.
+
+✔ How to use Foundry to access contract's level data.
+
+✔ ️How to use Foundry to deploy your contracts on an Ethereum testnet.
+
+## Introduction
+
+Ethereum has introduced a major revolution in the blockchain world: smart-contracts.
+Smart-contracts rely on blockchain technology to guarantee their integrity and inviolability.
+They are irrevocable computer programs, most often deployed on a blockchain,
+which execute a set of pre-defined instructions. Developing smart-contracts is way more different 
+from developing a regular application.
+
+## Step 0 - Setup
+
+Please refer to the [SETUP.md](./SETUP.md) file.
+
+## Step 1 - Forge is useful to test
+
+### :bookmark_tabs: **Description**:
+
+In this step you will learn how to use foundry to test your contracts.
+Before starting take a look at the generated files, specially in the `src/` & `test/` folders.
+In the `src/` folder you will find the `Counter.sol` file, this is the contract you will be working on.
+
+
+
+### :pushpin: **Tasks**:
+
+- Add a constructor to the contract, it will be used to initialize the number value. 
+    - The constructor must take a `uint256` as an argument.
+
+
+- Modify the `increment` function to make it increase the value of `number` by the value passed as an argument.
+    - The function must also take a `uint256` as an argument.
+
+  
+- Now test your function by modifying the `test/Counter.t.sol` file, the file must test the `increment` function and the `setNumber` one.
+    - You can use the `forge test` command at the root of your project to test your contract.
+
+> :attention: Take care about the visibility of your functions !
+
+### :books: **Documentation**:
+
+- [functions and constructor](https://docs.soliditylang.org/en/v0.8.13/contracts.html)
+- [test with forge](https://book.getfoundry.sh/forge/writing-tests)
+- [visibility](https://docs.soliditylang.org/en/v0.8.13/contracts.html#function-visibility)
+
+### ✔️ **Validation**:
+
+```shell
+Running 2 tests for test/Counter.t.sol:CounterTest
+[PASS] testIncrement() (gas: 28312)
+[PASS] testSetNumber(uint256) (runs: 256, μ: 26676, ~: 28387)
+Test result: ok. 2 passed; 0 failed; finished in 12.26ms
+```
+
+## Step 2 - Forge is also useful to deploy & call is useful to interact with contracts
```

Comment:
Step names should be concise and clear imo, you can detail later in the step description